### PR TITLE
Improved the intial matcher

### PR DESCRIPTION
**Pull request by aviralch**
Accounted for singular and plural. 

Broke up the code into functions for more readability 

Need to set up docker, to handle environment more easily

### CODE REVIEW

File: api/parse.py [hunk 1]
```diff
@@ -32,111 +33,101 @@ def food_database():
     meat = df["meat"]
     fruit_veg = df["fruit_veg"]
     food_groups = np.concatenate((dairy, meat, grain, fruit_veg))
-    # print(type(food_groups))
-    #return dairy, grain, meat, fruit_veg
     return food_groups
 
+# determines if a string is a float or an integer 
+def float_or_int(string):
```
Comment (kevmo314) [inline api/parse.py]: `str.isnumeric()` I think does what you're looking for? https://docs.python.org/3/library/stdtypes.html#str.isnumeric

File: api/parse.py [hunk 2]
```diff
@@ -32,111 +33,101 @@ def food_database():
...
+    for i in range(len(matches)):    
+        match_id, start, end = matches [0]
+        # results.append((nlp.vocab.strings[match_id],text_doc_plural[start:end]))
+        results.append(text_doc_plural[start:end])
 
-# print(data.get('ParsedResults')[0]['TextOverlay']['Lines'][22].get('LineText'))
-# print(i.get('LineText'))
-# print((new_data["ParsedResults"][0].values()))
-# filtered_data = (new_data["ParsedResults"][0].get('LineText'))
+    results = dict((i, results.count(i)) for i in results)
```
Comment (kevmo314) [inline api/parse.py]: `results = {i: results.count(i) for i in results}`

File: api/parse.py [hunk 3]
```diff
@@ -32,111 +33,101 @@ def food_database():
...
     food_groups = food_database()
-    fruit_veg_string = food_groups[3]]
 
     # Building the intial list of keywords we want to match the parsed items aganist
     patterns = [nlp(text) for text in food_groups]
-    matcher.add("TerminologyList", None, *patterns)
-
-    text_doc = nlp(words_string)
-    matches = matcher(text_doc)
+    matcher.add("Food Matcher", None, *patterns)
```
Comment (kevmo314) [inline api/parse.py:97]: Maybe pull the result stuff out into a separate function? Something like:

```
def get_results(text_doc_singular, text_doc_plural, matcher):
  results = []
  matches = matcher(text_doc_singular)
  for i in range(...):
    results.append(...)
  return results
```

File: api/parse.py [hunk 4]
```diff
@@ -32,111 +33,101 @@ def food_database():
...
-    fruit_veg_string = food_groups[3]]
 
     # Building the intial list of keywords we want to match the parsed items aganist
     patterns = [nlp(text) for text in food_groups]
-    matcher.add("TerminologyList", None, *patterns)
-
-    text_doc = nlp(words_string)
-    matches = matcher(text_doc)
+    matcher.add("Food Matcher", None, *patterns)    
+    matches = matcher(text_doc_singular)
```
Comment (kevmo314) [inline api/parse.py]: What about `matches(text_doc_plural)`?

**Review by kevmo314 — APPROVED** [review]

File: api/parse.py (hunk 1 above)
Comment (aviralch) [reply api/parse.py]: thanks, that was what I was looking for

File: api/parse.py (hunk 4 above)
Comment (aviralch) [reply api/parse.py]: it's there after the for loop for singular matching

**Comment by Leeoku** [comment]
great job avi

**Pull request merged**